firehose/parsers/flawfinder.py

```diff
@@ -0,0 +1,162 @@
+
+    flawfinder_data = []
+    pattern = "\([a-z]*\)"
+    pattern2 = "\[[0-9]\]"
+    prog = re.compile(pattern)
+    prog2 = re.compile(pattern2)
+
+
+    while line:
+        if line != "\n" and (prog.search(line) and prog2.search(line)):
+            data = {}
+
+            data["cwe"] = ''
+            data["path"] = line.split(":")[0]
+            data["line"] = line.split(":")[1]
+
+            message_line = loaded_file.readline()
+            message = ""
+            while message_line != "\n" and not \
+            (prog.search(message_line) and prog2.search(message_line)):
+                message += " " + message_line
+                extracted_cwe = get_cwe(message_line)
+                if extracted_cwe != '':
+                    data["cwe"] = extracted_cwe
+                message_line = loaded_file.readline()
+            data["message"] = message;
+            line = message_line
+            flawfinder_data.append(data)
+        else:
+            line = loaded_file.readline()
+
+    for data in flawfinder_data:
+        location = Location(file=File(data["path"], None),
+                            function=None,
+                            point=Point(int(data["line"]), 0))
+
+        cwe = data["cwe"]
+        if cwe == '':
+            cwe = None
+        else:
+            cwe = int(data["cwe"])
+
+        issue = Issue(cwe, None, location,
+                      Message(text=data["message"]), notes=None,
+                      trace=None, severity=None, customfields=None)
+
+        analysis.results.append(issue)
+
+    return analysis
+
+
+def get_cwe(line):
+    """Extract CWE from flawfinder alarms
+    It's possible that a alarm not have a correlated CWE
+    :line: A line of the report
+    :returns: The number of the CWE in the line, if it exists
+
+    Flawfinder can correlates multiple cwes with a single issue.
+    For now, Firehose not supports multiple CWEs in its models.
+    To minimize this problem, when multiple cwes occurs in flawfinder report,
+    Firehose will only parse the first CWE.
+
+    A issue was reported on the repository about this.
+    https://github.com/fedora-static-analysis/firehose/issues/35
+
+    If we have on the report the CWEs (CWE-120,CWE-20),
+    get_cwe() will returns only 120, that is the number of the
+    first CWE.
+    """
+
+    cwe = ''
+    cwe_init_str_pos = line.find("CWE-")
+    if cwe_init_str_pos > 0:
+        """ cwe_first_int_char indicates the first char of
+            the numeric part of the cwe
+            CWE-120, cwe_int_part will point to the character 1
+        """
+        cwe_first_int_char = cwe_init_str_pos + 4
+        try:
+            while 1:
+                int(line[cwe_first_int_char])
+                cwe_first_int_char += 1
+        except Exception:
+            cwe = line[(cwe_init_str_pos + 4):(cwe_first_int_char)]
+
+    return cwe
+
+
+def flawfinder(first_line):
+    """Retrieve flawfinder version from report.
```

Comment:
I find the current name to be unclear; please rename to ``parse_first_line`` (as that's what it does).